.gitlab/ruby-build-and-test-extra.yml

```diff
@@ -37,6 +37,12 @@ intel_2022_1_0:
   allow_failure: true
   extends: .build_and_test_on_ruby
 
+intel_2022_1_0_mpi:
+  variables:
+    SPEC: "${PROJECT_RUBY_VARIANTS} +mpi %intel@2022.1.0 ${PROJECT_RUBY_DEPS} ^mvapich2"
+  allow_failure: true
```

Comment:
Did you intend to allow this to fail?

Reply:
Why is intel_2022_1_0 allowed to fail in the first place?

Reply:
dunno. maybe adrien ran into issues. i'll have to take a look. let's see what happens with this. we could enable mpi on the other job since they mostly overlap.

Reply:
Sounds good. WOuld I need to modify RUBY_BUILD_AND_TEST_JOB_ALLOC to get two ranks?

Reply:
Not sure. I'll ping adrien.

Reply:
@MrBurmark the `$RUBY_BUILD_AND_TEST_<JOB|SHARED>_ALLOC` only specifies a "macro" allocation: allocating nodes on the machine for a given time.

Under that allocation, the `scripts/gitlab/build_and_tests.sh` script is called. This script should not run on 2 ranks simultaneously. Instead, the script itself is calling the test suite, which may contain `srun` commands that invoke 2 ranks.

However, we do need to allow nested allocations with the `--overlap` option in `$RUBY_BUILD_AND_TEST_<JOB|SHARED>_ALLOC`.